auth/authentication.go

```diff
@@ -26,47 +26,56 @@ type Auth struct {
 	DB Database
 }
 
-type authenticate func(tokenID string, user *model.User) (authenticated bool, success bool, userId uint)
+type authenticate func(tokenID string, user *model.User) (authenticated bool, success bool, userId uint, err error)
 
 // RequireAdmin returns a gin middleware which requires a client token or basic authentication header to be supplied
 // with the request. Also the authenticated user must be an administrator.
 func (a *Auth) RequireAdmin() gin.HandlerFunc {
-	return a.requireToken(func(tokenID string, user *model.User) (bool, bool, uint) {
+	return a.requireToken(func(tokenID string, user *model.User) (bool, bool, uint, error) {
 		if user != nil {
-			return true, user.Admin, user.ID
+			return true, user.Admin, user.ID, nil
 		}
-		if token, _ := a.DB.GetClientByToken(tokenID); token != nil {
-			user, _ := a.DB.GetUserByID(token.UserID)
-			return true, user.Admin, token.UserID
+		if token, err := a.DB.GetClientByToken(tokenID); err != nil {
+			return true, false, 0, err
```

Comment:
This should be `false, false, 0, err` first flag is authenticated which should be false here. Please check the other returns, in case of an error the user isn't authenticated.